plugins/pulp_rpm/plugins/importers/yum/listener.py

```diff
@@ -196,6 +196,7 @@ def download_succeeded(self, report):
                 return
             added_unit = self.sync.add_rpm_unit(self.metadata_files, unit)
```

Comment:
A problem I've noticed is that if an exception occurs between this call and ``self.sync.associate_rpm_unit``, the unit is in the database (as an orphan), but it might not be backed by a file. I _think_ things are checking to make sure the file exists before associating it, but maybe that's not true.

A second issue is after this happens, the progress report indicates the failure, but after a resync the files are not downloaded unless a ``--force-full`` sync is done. I'm not really sure if this is a problem since that option exists to fix problems like this.